### PR TITLE
Add paid

### PR DESCRIPTION
**Pull request by TheLillyleaf**

### CODE REVIEW

**Review by TheLillyleaf — COMMENTED** [review]
self check

File: Purchase/index.ts [hunk 1]
```diff
@@ -134,8 +134,10 @@ export namespace Purchase {
 		])
 	}
 
+
```
Comment (EliasEriksson) [inline Purchase/index.ts:137]: github gives automatic warning here. run `npm run lint`

File: Purchase/index.ts [hunk 2]
```diff
@@ -134,8 +134,10 @@ export namespace Purchase {
...
+	
 	export async function compileExpense(
-		compileData: { purchases: { buyer: string; purpose: string; amount: Amount }[] },
+		compileData: Record<string, { purpose: string; date: isoly.Date; amount: Amount }[]>,
```
Comment (EliasEriksson) [inline Purchase/index.ts:140]: do we maybe want to extract these types to some Creatable of something? they are used in the API quite a lot as well if i am not mistaken?

**Review by EliasEriksson — COMMENTED** [review]
Maybe its time to take the setp you have been talking about? should we make a new Folder on the top level in the model called repport with Expence and Receipt repports? then we could have the API request type be `Repport.Expense.Creatable` / `Repport.Receipt.Creatable` and response could be `Repport.Expense` / `Repport.Receipt`. we could also moteh the apropriate compile function into those type away from receipt / purchase(?)

**Review by TheLillyleaf — COMMENTED** [review]
self check

File: Report/Creatable.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { isly } from "isly"
+import { Expense as ReportExpense } from "./Expense"
+import { Receipt as ReportReceipt } from "./Receipt"
+
+export type Creatable = Creatable.Expense | Creatable.Receipt
+
+export namespace Creatable {
+	const type = isly.union(ReportExpense.Creatable.type, ReportReceipt.Creatable.type)
```
Comment (EliasEriksson) [inline Report/Creatable.ts:8]: dont forget the type argument
`isly.union<Creatable>(...)`

File: Payment/Expense/Paid.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { isoly } from "isoly"
+
+export interface Paid {
+	created: isoly.Date
+	issuer: string
+}
+
+export namespace Paid {
+	export function is(value: Paid | any): value is Paid {
```
Comment (EliasEriksson) [inline Payment/Expense/Paid.ts:9]: rewrite with isly

File: Report/Expense/index.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
+import { isly } from "isly"
+import * as PDFLib from "pdf-lib"
+import { Creatable as ExpenseCreatable } from "./Creatable"
+
+export interface Expense {
+	file: Uint8Array
```
Comment (EliasEriksson) [inline Report/Expense/index.ts:6]: Just making sure. this is supposed to be `Uint8Array` and not `File` ?

**Review by EliasEriksson — CHANGES_REQUESTED** [review]
only few small things

**Pull request closed without merging**